projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -462,6 +525,101 @@ def preview(
 
         Args:
             open: Open in your browser after creating (default: True)
+            standalone: Inline the report source in the HTML report file rather than loading via CDN (default: False)
             formatting: Sets the basic report styling
         """
-        self._save(self._preview_file.name, open=open, formatting=formatting)
+        self._save(self._preview_file.name, open=open, standalone=standalone, formatting=formatting)
+
+    ############################################################################
+    # Local served reports
+    def build(
+        self, path: NPath, formatting: t.Optional[ReportFormatting] = None, compress_assets: bool = False
+    ) -> None:
+        """Build a report which can be served by a local http server
+
+        Args:
+            path: File path to store the document
+            compress_assets: Compress user assets during report generation (default: True)
+            formatting: Sets the basic report styling
+        """
+
+        if osp.isdir(path):
+            rmtree(path)
```

Comment:
I think you may want to ensure that `path` is a relative path and a child of the current working directory to be safe - you can make the path and current dir as absolute, and ensure it's a child path